src/Tests/Microsoft.Diagnostics.Monitoring.Tool.UnitTests/Eggress/AzureBlob/AzureBlobEgressProviderTests.cs

```diff
@@ -0,0 +1,295 @@
+        {
+            _azuriteFixture.SkipTestIfNotAvailable();
+
+            // Arrange
+            TestOutputLoggerProvider loggerProvider = new(_outputHelper);
+            AzureBlobEgressProvider egressProvider = new(loggerProvider.CreateLogger<AzureBlobEgressProvider>());
+
+            BlobContainerClient containerClient = await ConstructBlobContainerClientAsync();
+
+            BlobSasBuilder sasBuilder = new(BlobContainerSasPermissions.Add | BlobContainerSasPermissions.Create, DateTimeOffset.MaxValue);
+            Uri sasUri = containerClient.GenerateSasUri(sasBuilder);
+
+            AzureBlobEgressProviderOptions providerOptions = ConstructEgressProviderSettings(containerClient, sasToken: sasUri.Query);
+            EgressArtifactSettings artifactSettings = ConstructArtifactSettings();
+
+            // Act
+            string identifier = await egressProvider.EgressAsync(providerOptions, UploadAction, artifactSettings, CancellationToken.None);
+
+            // Assert
+            List<BlobItem> blobs = await GetAllBlobsAsync(containerClient);
+            Assert.Single(blobs);
+
+            BlobItem resultingBlob = blobs.First();
+            Assert.Equal($"{providerOptions.BlobPrefix}/{artifactSettings.Name}", resultingBlob.Name);
+        }
+
+        [ConditionalFact(Timeout = TestTimeouts.EgressUnitTestTimeoutMs)]
+        public async Task AzureBlobEgress_ThrowsWhen_ContainerDoesNotExistAndUsingRestrictiveSasToken()
+        {
+            _azuriteFixture.SkipTestIfNotAvailable();
+
+            // Arrange
+            TestOutputLoggerProvider loggerProvider = new(_outputHelper);
+            AzureBlobEgressProvider egressProvider = new(loggerProvider.CreateLogger<AzureBlobEgressProvider>());
+
+            BlobContainerClient containerClient = await ConstructBlobContainerClientAsync(create: false);
+
+            BlobSasBuilder sasBuilder = new(BlobContainerSasPermissions.Add, DateTimeOffset.MaxValue);
+            Uri sasUri = containerClient.GenerateSasUri(sasBuilder);
+
+            AzureBlobEgressProviderOptions providerOptions = ConstructEgressProviderSettings(containerClient, sasToken: sasUri.Query);
+            EgressArtifactSettings artifactSettings = ConstructArtifactSettings();
+
+            // Act & Assert
+            await Assert.ThrowsAsync<EgressException>(async () => await egressProvider.EgressAsync(providerOptions, UploadAction, artifactSettings, CancellationToken.None));
+        }
+
+        private Task<Stream> UploadAction(CancellationToken token)
+        {
+            return Task.FromResult<Stream>(new FileStream(_testUploadFile, FileMode.Open, FileAccess.Read));
+        }
+
+        private async Task<BlobContainerClient> ConstructBlobContainerClientAsync(string containerName = null, bool create = true)
+        {
+            BlobServiceClient serviceClient = new(_azuriteFixture.Account.ConnectionString);
+
+            containerName ??= Guid.NewGuid().ToString("D");
```

Comment:
If a container name isn't specified, it should go to the root of the account, which I believe has a sentinel container name value of `$root`.